stan/math/prim/err/elementwise_check.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef STAN_MATH_PRIM_ERR_ELEMENTWISE_ERROR_CHECKER_HPP
+#define STAN_MATH_PRIM_ERR_ELEMENTWISE_ERROR_CHECKER_HPP
+
+#include <stan/math/prim/err/throw_domain_error.hpp>
+#include <stan/math/prim/fun/get.hpp>
+#include <stan/math/prim/fun/size.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+#include <string>
+#include <sstream>
+#include <vector>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+template <typename T_is_good, typename T_exception>
+struct Checker {
+  const T_is_good& is_good;
+  const char* function;
+  const char* name;
+  const char* suffix;
+
+  void raise_error_impl(std::stringstream& ss) { throw T_exception{ss.str()}; }
```

Comment:
Functions also need doc.  

Also, `_impl` should be something like `_stream` or `_ss` to give readers a hint at how this function is different than `raise_error`.

Reply:
Fixed. Its _ss now.